mifosng-android/src/main/java/com/mifos/mifosxdroid/login/LoginActivity.java

```diff
@@ -171,32 +184,36 @@ public void onLoginSuccessful(User user) {
 
     @Override
     public void onLoginError(Throwable throwable) {
-        try {
-
-            if (throwable.getCause() instanceof SSLHandshakeException) {
-                promptUserToByPassTheSSLHandshake();
-            } /*else if (throwable.getResponse().getStatus() ==
-                    org.apache.http.HttpStatus.SC_UNAUTHORIZED) {
-                Toaster.show(findViewById(android.R.id.content), getString(R.string
-                        .error_login_failed));
-            } else if (retrofitError.getResponse().getStatus() ==
-                    org.apache.http.HttpStatus.SC_INTERNAL_SERVER_ERROR) {
-                Toaster.show(findViewById(android.R.id.content), "Internal server error");
-            }*/
-        } catch (NullPointerException e) {
-            if (Network.getConnectivityStatusString(this).equals("Not connected to " +
-                    "Internet")) {
-                Toaster.show(findViewById(android.R.id.content), "Not connected to Network");
-            } else {
-                Toaster.show(findViewById(android.R.id.content), getString(R.string.error_unknown));
-            }
+
+        if (throwable instanceof SSLHandshakeException) {
```

Comment:
We are using Retrofit 2, you can cast ```Throwable``` with ```HttpException``` and access failed errors.
For example look here https://github.com/openMF/android-client/blob/master/mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/identifierdialog/IdentifierDialogPresenter.java#L93